### ISSUE TITLE
Getting AURemoteIO::IOThread when implementing more than one conductor / sustainer

### ISSUE THREAD

**Issue by RocKhalil** [AudioKit/AudioKitSynthOne#21]
I created a fork of SynthOne app and I'm trying to load multiple presets at the same time.

Right now, I'm creating a list of `conductors` that contains `sustainers` and each sustainer loads a preset. This is working perfectly on the simulator. 2 and more presets are running at the same time. However, when I try using the app on my iPhone 7 Plus, I get AURemoteIO::IOThread.

Any idea why this is happening ? Maybe memory leaks ?

--
Controller:

```
import UIKit
import AudioKit
import CoreMotion

class MyViewController: UIViewController, AKMIDIListener {
    let channels = 2
    
    var conductors = [Conductor]()
    var sustainers = [SDSustainer]()

    var appSettings = AppSettings()

    var presetsA = [Preset]()
    var presetsB = [Preset]()

    var playingPresets: [Int: Preset] = [:]
    var playing = false

    var presetsInitialized = false

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.

        DispatchQueue.global(qos: .userInteractive).async {
            AudioKit.midi.createVirtualInputPort(95_433, name: "ASO")
            AudioKit.midi.openInput()
            AudioKit.midi.openOutput("AudioKit Synth One")
        }
        AudioKit.midi.addListener(self)

        var callbackStruct = AudioOutputUnitMIDICallbacks(
            userData: nil,
            MIDIEventProc: { (_, status, data1, data2, _) in
                AudioKit.midi.sendMessage([MIDIByte(status), MIDIByte(data1), MIDIByte(data2)])
            },
            MIDISysExProc: { (_, _, _) in
                print("Not handling sysex")
            }
        )

        guard let outputAudioUnit = AudioKit.engine.outputNode.audioUnit else {
            AKLog("ERROR: can't create outputAudioUnit")
            return
        }

        let connectIAAMDI = AudioUnitSetProperty(outputAudioUnit,
                                                 kAudioOutputUnitProperty_MIDICallbacks,
                                                 kAudioUnitScope_Global,
                                                 0,
                                                 &callbackStruct,
                                                 UInt32(MemoryLayout<AudioOutputUnitMIDICallbacks>.size))
        if connectIAAMDI != 0 {
            AKLog("Something bad happened")
        }
    }
    
    func initializeMusicPlaying() {
        for _ in (0..<channels){
            let conductor = Conductor()
            conductor.start()
            let sustainer = SDSustainer(conductor.synth)

            conductors.append(conductor)
            sustainers.append(sustainer)
        }
        
        presetsA = loadBank(fileName: "A– Drums", conductor: conductors[0])
        presetsB = loadBank(fileName: "B– Leads", conductor: conductors[1])

        var loaded = false
        while !loaded { loaded = loadPreset(preset: presetsA[0], index: 0) }

        loaded = false
        while !loaded { loaded = loadPreset(preset: presetsB[1], index: 1) }

        presetsInitialized = true
    }
    
    func loadBank(fileName: String, conductor: Conductor) -> [Preset]
    {
        var presets = [Preset]()
        guard let jsonArray = getJSON(fileName: fileName, extn: "json") else { return [] }
        
        for preset in jsonArray{
            presets.append(Preset(dictionary: preset as! [String : Any], conductor: conductor))
        }
        
        return presets
    }
    
    func getJSON(fileName: String, extn: String) -> [Any]?
    {
        if let path = Bundle.main.path(forResource: fileName, ofType: extn) {
            do {
                let data = try Data(contentsOf: URL(fileURLWithPath: path), options: .mappedIfSafe)
                let jsonResult = try JSONSerialization.jsonObject(with: data, options: .mutableLeaves)
                return jsonResult as? [Any]
            } catch {
                // handle error
                return []
            }
        }
        
        return []
    }
    
    func loadPreset(preset: Preset, index: Int) -> Bool {
        let conductor = conductors[index]

        guard let s = conductor.synth else {
            print("ERROR:can't load preset if synth is not initialized")
            return false
        }

        // The DEV panel has toggles (stored in settings) that impact loading of subsets of parameters of a Preset

        if !appSettings.freezeArpRate {
            s.setSynthParameter(.arpRate, preset.arpRate)
        }

        if !appSettings.freezeDelay {
            s.setSynthParameter(.delayOn, preset.delayToggled)
            s.setSynthParameter(.delayFeedback, preset.delayFeedback)
            s.setSynthParameter(.delayMix, preset.delayMix)
            s.setSynthParameter(.delayTime, preset.delayTime)
            s.setSynthParameter(.delayInputCutoffTrackingRatio, preset.delayInputCutoffTrackingRatio)
            s.setSynthParameter(.delayInputResonance, preset.delayInputResonance)
        }

        if !appSettings.freezeReverb {
            s.setSynthParameter(.reverbOn, preset.reverbToggled)
            s.setSynthParameter(.reverbFeedback, preset.reverbFeedback)
            s.setSynthParameter(.reverbHighPass, preset.reverbHighPass)
            s.setSynthParameter(.reverbMix, preset.reverbMix)
        }

        if !appSettings.freezeArpSeq {
            s.setSynthParameter(.arpIsOn, preset.isArpMode)
            s.setSynthParameter(.arpIsSequencer, preset.arpIsSequencer ? 1 : 0 )
            for i in 0..<16 {
                s.setPattern(forIndex: i, preset.seqPatternNote[i])
                s.setOctaveBoost(forIndex: i, preset.seqOctBoost[i] ? 1 : 0)
                s.setNoteOn(forIndex: i, preset.seqNoteOn[i])
            }
        }

        s.setSynthParameter(.tempoSyncToArpRate, preset.tempoSyncToArpRate)
        s.setSynthParameter(.lfo1Rate, preset.lfoRate)
        s.setSynthParameter(.lfo2Rate, preset.lfo2Rate)
        s.setSynthParameter(.delayTime, preset.delayTime)
        s.setSynthParameter(.autoPanFrequency, preset.autoPanFrequency)
        s.setSynthParameter(.masterVolume, preset.masterVolume)
        s.setSynthParameter(.isMono, preset.isMono)
        s.setSynthParameter(.glide, preset.glide)
        s.setSynthParameter(.widen, preset.widen)
        s.setSynthParameter(.index1, preset.waveform1)
        s.setSynthParameter(.index2, preset.waveform2)
        s.setSynthParameter(.morph1SemitoneOffset, preset.vco1Semitone)
        s.setSynthParameter(.morph2SemitoneOffset, preset.vco2Semitone)
        s.setSynthParameter(.morph2Detuning, preset.vco2Detuning)
        s.setSynthParameter(.morph1Volume, preset.vco1Volume)
        s.setSynthParameter(.morph2Volume, preset.vco2Volume)
        s.setSynthParameter(.morphBalance, preset.vcoBalance)
        s.setSynthParameter(.subVolume, preset.subVolume)
        s.setSynthParameter(.subOctaveDown, preset.subOsc24Toggled)
        s.setSynthParameter(.subIsSquare, preset.subOscSquareToggled)
        s.setSynthParameter(.fmVolume, preset.fmVolume)
        s.setSynthParameter(.fmAmount, preset.fmAmount)
        s.setSynthParameter(.noiseVolume, preset.noiseVolume)
        s.setSynthParameter(.cutoff, preset.cutoff)
        s.setSynthParameter(.resonance, preset.resonance)
        s.setSynthParameter(.filterADSRMix, preset.filterADSRMix)
        s.setSynthParameter(.filterAttackDuration, preset.filterAttack)
        s.setSynthParameter(.filterDecayDuration, preset.filterDecay)
        s.setSynthParameter(.filterSustainLevel, preset.filterSustain)
        s.setSynthParameter(.filterReleaseDuration, preset.filterRelease)
        s.setSynthParameter(.attackDuration, preset.attackDuration)
        s.setSynthParameter(.decayDuration, preset.decayDuration)
        s.setSynthParameter(.sustainLevel, preset.sustainLevel)
        s.setSynthParameter(.releaseDuration, preset.releaseDuration)
        s.setSynthParameter(.bitCrushSampleRate, preset.crushFreq)
        s.setSynthParameter(.autoPanAmount, preset.autoPanAmount)
        s.setSynthParameter(.reverbOn, preset.reverbToggled)
        s.setSynthParameter(.reverbFeedback, preset.reverbFeedback)
        s.setSynthParameter(.reverbHighPass, preset.reverbHighPass)
        s.setSynthParameter(.reverbMix, preset.reverbMix)
        s.setSynthParameter(.delayOn, preset.delayToggled)
        s.setSynthParameter(.delayFeedback, preset.delayFeedback)
        s.setSynthParameter(.delayMix, preset.delayMix)
        s.setSynthParameter(.lfo1Index, preset.lfoWaveform)
        s.setSynthParameter(.lfo1Amplitude, preset.lfoAmplitude)
        s.setSynthParameter(.lfo2Index, preset.lfo2Waveform)
        s.setSynthParameter(.lfo2Amplitude, preset.lfo2Amplitude)
        s.setSynthParameter(.cutoffLFO, preset.cutoffLFO)
        s.setSynthParameter(.resonanceLFO, preset.resonanceLFO)
        s.setSynthParameter(.oscMixLFO, preset.oscMixLFO)
        s.setSynthParameter(.reverbMixLFO, preset.reverbMixLFO)
        s.setSynthParameter(.decayLFO, preset.decayLFO)
        s.setSynthParameter(.noiseLFO, preset.noiseLFO)
        s.setSynthParameter(.fmLFO, preset.fmLFO)
        s.setSynthParameter(.detuneLFO, preset.detuneLFO)
        s.setSynthParameter(.filterEnvLFO, preset.filterEnvLFO)
        s.setSynthParameter(.pitchLFO, preset.pitchLFO)
        s.setSynthParameter(.bitcrushLFO, preset.bitcrushLFO)
        s.setSynthParameter(.tremoloLFO, preset.tremoloLFO)
        s.setSynthParameter(.arpDirection, preset.arpDirection)
        s.setSynthParameter(.arpInterval, preset.arpInterval)
        s.setSynthParameter(.arpOctave, preset.arpOctave)
        s.setSynthParameter(.arpTotalSteps, preset.arpTotalSteps )
        s.setSynthParameter(.monoIsLegato, preset.isLegato )
        s.setSynthParameter(.phaserMix, preset.phaserMix)
        s.setSynthParameter(.phaserRate, preset.phaserRate)
        s.setSynthParameter(.phaserFeedback, preset.phaserFeedback)
        s.setSynthParameter(.phaserNotchWidth, preset.phaserNotchWidth)
        s.setSynthParameter(.filterType, preset.filterType)
        s.setSynthParameter(.compressorMasterRatio, preset.compressorMasterRatio)
        s.setSynthParameter(.compressorReverbInputRatio, preset.compressorReverbInputRatio)
        s.setSynthParameter(.compressorReverbWetRatio, preset.compressorReverbWetRatio)
        s.setSynthParameter(.compressorMasterThreshold, preset.compressorMasterThreshold)
        s.setSynthParameter(.compressorReverbInputThreshold, preset.compressorReverbInputThreshold)
        s.setSynthParameter(.compressorReverbWetThreshold, preset.compressorReverbWetThreshold)
        s.setSynthParameter(.compressorMasterAttack, preset.compressorMasterAttack)
        s.setSynthParameter(.compressorReverbInputAttack, preset.compressorReverbInputAttack)
        s.setSynthParameter(.compressorReverbWetAttack, preset.compressorReverbWetAttack)
        s.setSynthParameter(.compressorMasterRelease, preset.compressorMasterRelease)
        s.setSynthParameter(.compressorReverbInputRelease, preset.compressorReverbInputRelease)
        s.setSynthParameter(.compressorReverbWetRelease, preset.compressorReverbWetRelease)
        s.setSynthParameter(.compressorMasterMakeupGain, preset.compressorMasterMakeupGain)
        s.setSynthParameter(.compressorReverbInputMakeupGain, preset.compressorReverbInputMakeupGain)
        s.setSynthParameter(.compressorReverbWetMakeupGain, preset.compressorReverbWetMakeupGain)
        s.setSynthParameter(.delayInputCutoffTrackingRatio, preset.delayInputCutoffTrackingRatio)
        s.setSynthParameter(.delayInputResonance, preset.delayInputResonance)
        s.setSynthParameter(.pitchbendMinSemitones, preset.pitchbendMinSemitones)
        s.setSynthParameter(.pitchbendMaxSemitones, preset.pitchbendMaxSemitones)
        s.setSynthParameter(.frequencyA4, preset.frequencyA4)

        s.resetSequencer()

        playingPresets[index] = preset

        return true
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func playPreset(sustainer: SDSustainer, preset: Preset){
        sustainer.stopAll()

        preset.notes.forEach { note in
            sustainer.play(noteNumber: 60, velocity: 80)
        }
    }
    
    @IBAction func playMusic(_ sender: Any) {
        if(!presetsInitialized){
            initializeMusicPlaying()
        }

        if(playing){
            for i in (0..<channels){
                sustainers[i].stopAll()
            }

            playing = false
        }else{
            for i in (0..<channels){
                playPreset(sustainer: sustainers[i], preset: playingPresets[i]!)
            }

            playing = true
        }
    }
    
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
```

So basically, I'm creating an X amount of conductors, each has it own sustainer and each sustainer is loading a preset and playing it with a note

**Comment by aure** [comment]
Hi, sorry I haven't been watching the issues on SynthOne, but you can't create multiple conductor the way it is architected currently.  This will change with upcoming versions of AudioKit.

**Comment by RocKhalil** [comment]
Hey @aure, yup I figured that out;
I removed the multi-conductor and implemented a way to play multiple presets at the same time.
Running multiple conductors takes all the phone's memory. This is why it was crashing.
